CHANGELOG.md

```diff
@@ -1,3 +1,47 @@
+- **There are several changes to the behavior of scheduling and lifecycle methods:**
+  * `ReactDOM.render()` and `ReactDOM.unstable_renderIntoContainer()` now return `null` if called from inside a lifecycle method.
+    * To work around this, you can either use [the new portal API](https://github.com/facebook/react/issues/10309#issuecomment-318433235) or [refs](https://github.com/facebook/react/issues/10309#issuecomment-318434635).
+  * `setState`:
+    * Calling `setState` with null no longer triggers an update. This allows you to decide in an updater function if you want to re-render.
+    * Calling `setState` directly in render always causes an update. This was not previously the case. Regardless, you should not be calling setState from render.
+    * `setState` callback (second argument) now fires immediately after `componentDidMount` / `componentDidUpdate` instead of after all components have rendered.
+  * When replacing `<A />` with `<B />`,  `B.componentWillMount` now always happens before  `A.componentWillUnmount`. Previously, `A.componentWillUnmount` could fire first in some cases.
+  * Previously, changing the ref to a component would always detach the ref before that component's render is called. Now, we change the ref later, when applying the changes to the DOM.
+  * It is not safe to re-render into a container that was modified by something other than React. This worked previously in some cases but was never supported. We now emit a warning in this case. Instead you should clean up your component trees using `ReactDOM.unmountComponentAtNode`. [See this example.](https://github.com/facebook/react/issues/10294#issuecomment-318820987)
+  * `componentDidUpdate` lifecycle no longer receives `prevContext` param. (See #8631)
+  * Shallow renderer no longer calls `componentDidUpdate()` because DOM refs are not available. This also makes it consistent with `componentDidMount()` (which does not get called in previous versions either).
+  * Shallow renderer does not implement `unstable_batchedUpdates()` anymore.
+- **The names and paths to the single-file browser builds have changed to emphasize the difference between development and production builds.** For example:
+  - react/dist/react.js → react/umd/react.development.js
+  - react/dist/react.min.js → react/umd/react.production.min.js
+  - react-dom/dist/react-dom.js → react-dom/umd/react-dom.development.js
+  - react-dom/dist/react-dom.min.js → react-dom/umd/react-dom.production.min.js
+* ReactDOM.createPortal() is official and stable
+  ([@gaearon](https://github.com/gaearon) in
+  [10675](https://github.com/facebook/react/pull/10675))
+* Allow [passing some "unknown" attributes through to DOM components](https://facebook.github.io/react/blog/2017/09/08/dom-attributes-in-react-16.html) (@nhunzaker in https://github.com/facebook/react/pull/10385 , https://github.com/facebook/react/pull/10564 , https://github.com/facebook/react/pull/10495 and others)
```

Comment:
This also looks pretty bad in Markdown file. These need to be turned into real links.